master/internal/command/command.go

```diff
@@ -323,14 +323,14 @@ func (c *Command) ToV1Command() *commandv1.Command {
 func (c *Command) ToV1Notebook() *notebookv1.Notebook {
 	c.mu.Lock()
 	defer c.mu.Unlock()
-
 	allo := c.refreshAllocationState()
+	notebookAddress := fmt.Sprintf("%s?token=%s", c.serviceAddress(), c.Base.UserSessionToken)
```

Comment:
I'm a little worried about returning user tokens in ```/api/v1/notebooks``` for security reasons

is there a way around this?

Reply:
we have to give the plain text link with the token to the user, so they could copy-paste it into their vscode, right? so probably no way to avoid sending these secrets over the wire. 

we could maybe send the base URL and the token separately, so at least the automated URL scanners could not catch it. but that is security by obscurity.

Reply:
I think my issue is we are exposing Determined users tokens to other users. An non admin could snag an admins token from the api and just start making calls as that user. Or you could impersonate another user.

I have mentioned at least part of this in a DM but an alternative that doesn't expose more access into the system is

1. Notebooks links in the API stay the same. When connecting from the CLI we add ```?userToken=token``` to the link we print. This token comes from being logged into the CLI or webui. 

2. The proxy authenticates the request based on ```?userToken=``` if present.

3. If vscode needs the notebook to be authenticated the proxy can add another layer of auth between itself and the notebook. This is optional to me since we don't assume this connection is authed today. But I don't think the notebook could use the userToken as the token it expects because multiple tokens should be valid to access the notebook.

Reply:
oh, good point.